File: 4-median-of-two-sorted-arrays/median-of-two-sorted-arrays.java
class Solution {
    public double findMedianSortedArrays(int[] nums1, int[] nums2) {
        double ans = 0;
        int n = nums1.length;
        int m = nums2.length;
        int[] arr = new int[n+m];
        int k=0;
        for(int i=0;i<n;i++) {
            arr[k] = nums1[i];
            k+=1;
        }
        for(int i=0;i<m;i++) {
            arr[k] = nums2[i];
            k+=1;
        }
        Arrays.sort(arr);
       int len = arr.length; 
       if(len%2!=0) {
            int val = (int)len/2;
            ans = arr[val];
       }
       else {
         int val = (int) len/2;
         ans =(double) (arr[val]+arr[val-1])/2;     
       }
       return ans;
    }
}

File: 4-median-of-two-sorted-arrays/median-of-two-sorted-arrays.cpp
class Solution {
public:
    double findMedianSortedArrays(vector<int>& nums1, vector<int>& nums2) {
        int n = nums1.size();
        int m = nums2.size();
        nums1.resize(n+m);
        for(int i=0;i<m;++i) {
          nums1[n+i] = nums2[i];
        }
        sort(nums1.begin(),nums1.end());
        n = nums1.size();
        if(n % 2 == 0) {
            int mid = n / 2;
            double ans = (nums1[mid]+nums1[mid-1])/2.0;
            return (double)ans;
        }
        return (double)nums1[n/2];

    }
};